imagepuller/internal/service/verify_test.go

```diff
@@ -102,3 +118,225 @@ func TestStoreAndVerifyLayers_EvilRegistry(t *testing.T) {
 		})
 	}
 }
+
+type StubRemote struct {
+	headErr  bool
+	imageErr bool
+	indexErr bool
+
+	DefaultRemote
```

Comment:
I agree with @katexochen that this delegation here is unexpected and complicates the test. On the other hand, faking the interfaces provided by the library would require significant amounts of code, so I would rather take the evil registry.

Did you consider https://pkg.go.dev/github.com/google/go-containerregistry/pkg/v1/fake? That might be the sweet spot in between, but I did not check how easy it is to use.